File: lib/qt5/RegionListComboBox.cpp
/*
 * Copyright (c) 2016 Nicholas Corgan (n.corgan@gmail.com)
 *
 * Distributed under the MIT License (MIT) (See accompanying file LICENSE.txt
 * or copy at http://opensource.org/licenses/MIT)
 */

#include <pkmn/qt5/RegionListComboBox.hpp>

#include <pkmn/database/lists.hpp>

namespace pkmn { namespace qt5 {

    RegionListComboBox::RegionListComboBox(
        QWidget* parent
    ): QComboBox(parent)
    {
        std::vector<std::string> regions = pkmn::database::get_region_list();
        for(auto iter = regions.begin(); iter != regions.end(); ++iter) {
            addItem(QString::fromStdString(*iter));
        }

        setEditable(false);
    }

}}

// Generated at build time
#include "../include/pkmn/qt5/moc_RegionListComboBox.cpp"

File: maven/PKMN/src/main/java/nc/PKMN/RegionListComboBox.java
/*
 * Copyright (c) 2016 Nicholas Corgan (n.corgan@gmail.com)
 *
 * Distributed under the MIT License (MIT) (See accompanying file LICENSE.txt
 * or copy at http://opensource.org/licenses/MIT)
 */

package nc.PKMN;

import javax.swing.JComboBox;

/**
 * A JComboBox filled with a list of regions in the main series and Gamecube games.
 */
public class RegionListComboBox extends JComboBox<String> {
    public RegionListComboBox() {
        StringVector regionList = Database.getRegionList();
        for(long i = 0; i < regionList.size(); i++) {
            this.addItem(regionList.get((int)i));
        }
        this.setEditable(false);
    }
}
